projects/httpProxy.c

```diff
@@ -51,19 +52,32 @@ int main(int argc, char* argv[]) {
 
     printf("HTTP Proxy Server Listening on port %d\n", PORT);
 
+    pthread_t threads[MAX_THREADS] = {0};
+    int current_thread = 0;
+
     while (1) {
         // Принятие соединения от клиента
         if ((client_socket = accept(server_fd, (struct sockaddr*)&address, (socklen_t*)&addr_len)) < 0) {
             perror("accept");
             continue;
         }
 
-        // Создание потока для обработки соединения
-        pthread_t thread_id;
-        if (pthread_create(&thread_id, NULL, handle_connection, (void*)&client_socket) != 0) {
+        if (threads[current_thread] != 0) {
+            pthread_join(threads[current_thread], NULL);
```

Comment:
You have one long-running download thread and all remaining are for short pieces of data.  As a result you'll stuck in waiting this long thread while all remaining transfers are finished long time ago.

Reply:
fixed,made threads detach